### PR TITLE
LPS-151510

### PR DESCRIPTION
**Pull request by holatuwol**
https://issues.liferay.com/browse/LPS-151510
https://issues.liferay.com/browse/PTR-2987

/cc @robertoDiaz 

### Related pull requests

From @ericyanLr :
> As discussed, this PR adds support for HEIC images to generate previews/thumbnails for Adaptive Media when ImageMagick (with HEIC support) is setup and enabled.

https://github.com/holatuwol/liferay-portal/pull/258
https://github.com/holatuwol/liferay-portal/pull/259

### Solution notes

The solution on https://github.com/holatuwol/liferay-portal/pull/258 modified the `AMImageScaledImage` interface to add the `getMimeType` method, which (due to being a `ConsumerType` interface) resulted in a major version change when running baseline, which we would not be able to backport to earlier branches. The solution on https://github.com/holatuwol/liferay-portal/pull/259 avoids the major version change by extracting the mime type again from the InputStream.

This pull request adapts the solution in https://github.com/holatuwol/liferay-portal/pull/258 with a new interface, `AMImageConvertedImage`, as an indicator that the image might have been converted and thus can have a different mime type from the original FileVersion. The value is then read by the underlying service, and the correct mime type is persisted, without requiring a major version change.

### Other alternative solutions

Other alternatives that were explored:

* Using the solution in https://github.com/holatuwol/liferay-portal/pull/258 but updating the existing `AMImageScaledImage` to be a `ProviderType`. This resolves the issue, but switching between the two service types is risky, because it breaks the existing API contract.

* Automatically converting images so that RenderedImage is only null when a conversion is not possible. While this addresses the issue, it also changes the API contract, where the returned ImageBag may have a different content type than the file that was passed in rather than returning null.

If there is a solution you believe we should implement, please let us know.

### CODE REVIEW

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

File: modules/apps/adaptive-media/adaptive-media-image-api/src/main/java/com/liferay/adaptive/media/image/scaler/AMImageConvertedImage.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+package com.liferay.adaptive.media.image.scaler;
+
+/**
+ * @author Minhchau Dang
+ */
+public interface AMImageConvertedImage {
+
+	public String getMimeType();
+
+}
```
Comment (adolfopa) [inline modules/apps/adaptive-media/adaptive-media-image-api/src/main/java/com/liferay/adaptive/media/image/scaler/AMImageConvertedImage.java:24]: @holatuwol Can we get rid of this interface? Why not add the getMimeType method to the AMImageScaledImage one?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/adaptive-media/adaptive-media-image-api/src/main/java/com/liferay/adaptive/media/image/scaler/AMImageConvertedImage.java (hunk 1 above)
Comment (ericyanLr) [reply modules/apps/adaptive-media/adaptive-media-image-api/src/main/java/com/liferay/adaptive/media/image/scaler/AMImageConvertedImage.java:24]: Hi @adolfopa ,

We considered adding this method, but found limitations especially for backports.

> **Solution notes**
> 
> The solution on https://github.com/holatuwol/liferay-portal/pull/258 modified the AMImageScaledImage interface to add the getMimeType method, which (due to being a ConsumerType interface) resulted in a major version change when running baseline, which we would not be able to backport to earlier branches.

**Pull request closed without merging**